docs/style-spec/expressions.md

```diff
@@ -0,0 +1,40 @@
+# Style Function Expressions
+
+**Constants:**
+- LN2
+- PI
+- e
```

Comment:
> By the way, one could make the same argument for treating bare strings in the JSON as constant values always and requiring the use of a `{ref: …}` object to represent a property reference. But that would either move the expression syntax away from the filter syntax or necessitate a backwards-incompatible change to the filter syntax.

Yes, that's exactly what @anandthakker and I are agreeing we should do (except I suggested `["data", …]` rather than `{ref: …}`).

In this proposed grammar, the first element of a JSON array is always a JSON string constant naming a style spec function. The zero or more other elements are expressions. One type of expression is a string constant expression, which is represented by a JSON string constant. In no case does a string constant alone denote "the value of the property with this name". This is indeed a break from the filter syntax.

This does mean that there can't be a direct correspondence between JSON array literals and style array literals. For example, an expression whose result is a font stack array will need to be something like `["array", "Open Sans Regular", "Arial Unicode MS Regular"]`.